bin/codemods/run.js

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env node
+
+/**
+ * External dependencies
+ */
+const fs = require( 'fs' );
+const path = require( 'path' );
+const child_process = require( 'child_process' );
+
+/**
+ * Internal dependencies
+ */
+const config = require( './src/config' );
+
+function getLocalCodemodFileNames() {
+	// Returns all JS files in bin/codemods/src folder, except for config and helpers.
+
+	return fs.readdirSync( './bin/codemods/src' )
+		.filter( name => name !== 'config.js' && name != 'helpers.js' ) // exclude utility files
+		.filter( name => name[ 0 ] !== '.' ) // exclude dot files
+		.filter( name => path.extname( name ) === '.js' ) // only include JS files
+		.map( name => name.replace( '.js', '' ) ); // strip extension from filename
+}
+
+function getValidCodemodNames() {
+	return [
+		...getLocalCodemodFileNames(),
+		...Object.getOwnPropertyNames( config.codemodArgs ),
+	].map( name => '- ' + name ).sort();
+}
+
+function printableValidCodemodNames() {
+	return getValidCodemodNames().join('\n');
+}
+
+function generateBinArgs( name ) {
+	if ( config.codemodArgs.hasOwnProperty( name ) ) {
+		// Is the codemod defined in the codemodArgs object?
+		return config.codemodArgs[ name ];
+	}
+
+	if ( getLocalCodemodFileNames().includes( name ) ) {
+		// Is the codemod a local script defined in bin/codemods/src folder?
+		return [
+			`--transform=bin/codemods/src/${ name }.js`,
+		];
+	}
+
+	throw new Error(
+		`"${ name }" is an unrecognized codemod.`
+	);
+}
+
+function runCodemod() {
+	const args = process.argv.slice( 2 );
+	if ( args.length === 0 || args.length === 1 ) {
+		process.stdout.write( '\n' );
+		process.stdout.write( './bin/codemods/run.js [transformation name] [target(s)]\n' );
+		process.stdout.write( '\n' );
+		process.stdout.write( `Valid transformation names: \n${ printableValidCodemodNames() }\n` );
+		process.stdout.write( '\n' );
+		process.stdout.write( 'Example: "./bin/codemods/run.js commonjs-imports client/blocks client/devdocs"\n' );
+		process.stdout.write( '\n' );
+		process.exit( 0 );
+	}
+
+	const [ name, ...targets ] = args;
+	const binArgs = [
+		// jscodeshift options
+		...config.jscodeshiftArgs,
+
+		...generateBinArgs( name ),
+
+		// Transform target
+		...targets,
+	];
+
+	const binPath = path.join( '.', 'node_modules', '.bin', 'jscodeshift' );
+	const jscodeshift = child_process.spawn( binPath, binArgs );
+
+	jscodeshift.stdout.on( 'data', ( data ) => {
+		process.stdout.write( data );
+	} );
+
+	jscodeshift.stderr.on( 'data', ( data ) => {
+		process.stderr.write( data );
+	} );
```

Comment:
fun fact! we can probably get away with just passing the functions here since there's one argument only…

```js
jscodeshift.stdout.on( 'data', process.stdout.write );
jscodeshift.stderr.on( 'data', process.stdout.write );
```

there's a chance we get an error due to `this` binding, but if that's the case the way it's written is fine but we can also remove some syntax noise if we prefer

```js
jscodeshift.stdout.on( 'data', data => process.stdout.write( data ) );
```

Reply:
Node streams have a [pipe](https://nodejs.org/api/stream.html#stream_readable_pipe_destination_options) method that could make this even simpler:
```js
jscodeshift.stderr.pipe( process.stderr, { end: false } );
```
The `end` option prevents closing the destination stream after the source stream ends.

Reply:
Ha! I knew I was missing something simple :)

Reply:
Or this:
```js
const jscodeshift = child_process.spawn( binPath, binArgs, { stdio: 'inherit' } );
```
See [here](https://nodejs.org/api/child_process.html#child_process_options_stdio). I'm not sure what happens when the child process stdio streams are closed. Does the parent stdio get closed, too?